File: jmolecules-axon/02-axonstyle-cqrs-es-bank/src/main/java/org/jmolecules/example/axonframework/domain/event/BankAccountCreatedEvent.java
package org.jmolecules.example.axonframework.domain.event;

import org.axonframework.serialization.Revision;
import org.jmolecules.event.annotation.DomainEvent;

@Revision("1")
@DomainEvent(namespace = "axon.bank", name = "BankAccountCreatedEvent")
public record BankAccountCreatedEvent(
  String accountId,
  int initialBalance
) {
  // empty
}

File: jmolecules-axon/03-onion-axon-cqrs-es-bank/src/main/kotlin/org/jmolecules/example/axonframework/bank/domain/bankaccount/event/BankAccountCreatedEvent.kt
package org.jmolecules.example.axonframework.bank.domain.bankaccount.event

import org.axonframework.serialization.Revision
import org.jmolecules.event.annotation.DomainEvent
import org.jmolecules.example.axonframework.bank.domain.bankaccount.type.AccountId
import org.jmolecules.example.axonframework.bank.domain.bankaccount.type.Balance

@Revision("1")
@DomainEvent(namespace = "axon.bank", name = "BankAccountCreatedEvent")
data class BankAccountCreatedEvent(
  val accountId: AccountId,
  val initialBalance: Balance
)
